lib/smf/CMakeLists.txt

```diff
@@ -0,0 +1,12 @@
+# SPDX-License-Identifier: Apache-2.0
+
+zephyr_interface_library_named(smf)
+
+target_include_directories(smf INTERFACE ${CMAKE_CURRENT_SOURCE_DIR})
+
+zephyr_library()
+
+zephyr_include_directories(.)
```

Comment:
public API should likely be placed in `include/`, so this would not be required.

Reply:
Done